src/librustc_data_structures/obligation_forest/mod.rs

```diff
@@ -457,25 +655,59 @@ impl<O: ForestObligation> ObligationForest<O> {
             return Outcome {
                 completed: if do_completed == DoCompleted::Yes { Some(vec![]) } else { None },
                 errors,
-                stalled,
             };
         }
 
         self.mark_successes();
         self.process_cycles(processor);
         let completed = self.compress(do_completed);
 
-        Outcome { completed, errors, stalled }
+        Outcome { completed, errors }
+    }
+
+    #[inline(never)]
```

Comment:
Comment: what is the role of this function in the larger computation? What is a "blocked' node and when would it become unblocked? What variables of self are modified by this function?